src/scripts/extensions/safari/safariWorker.ts

```diff
@@ -129,34 +129,37 @@ export class SafariWorker extends ExtensionWorkerBase<SafariBrowserTab, SafariBr
 		});
 	}
 
-	private launchSafariPopupAndWaitForClose(url: string, autoCloseDestinationUrl: string): Promise<boolean> {
+	private launchSafariPopup(url: string, autoCloseDestinationUrl: string, waitForClose = false): Promise<boolean> {
 		return new Promise<boolean>((resolve, reject) => {
-			let redirectOccurred = false;
 			let newWindow = safari.application.openBrowserWindow();
 			newWindow.activeTab.url = url;
 
-			let errorObject;
-			let redirectListener = (event) => {
-				if (event && event.target && event.target.url && !event.target.url.toLowerCase().indexOf(autoCloseDestinationUrl)) {
-					redirectOccurred = true;
+			if (waitForClose) {
+				let redirectOccurred = false;
 
-					let error = Utils.getQueryValue(event.target.url, Constants.Urls.QueryParams.error);
-					let errorDescription = Utils.getQueryValue(event.target.url, Constants.Urls.QueryParams.errorDescription);
-					if (error || errorDescription) {
-						errorObject = { error: error, errorDescription: errorDescription };
-					}
+				let errorObject;
+				let redirectListener = (event) => {
+					if (event && event.target && event.target.url && !event.target.url.toLowerCase().indexOf(autoCloseDestinationUrl)) {
+						redirectOccurred = true;
+
+						let error = Utils.getQueryValue(event.target.url, Constants.Urls.QueryParams.error);
+						let errorDescription = Utils.getQueryValue(event.target.url, Constants.Urls.QueryParams.errorDescription);
+						if (error || errorDescription) {
+							errorObject = { error: error, errorDescription: errorDescription };
+						}
 
-					newWindow.removeEventListener(redirectListener);
-					newWindow.close();
-				}
-			};
-			newWindow.addEventListener("navigate", redirectListener);
-
-			let closeListener = (event) => {
-				errorObject ? reject(errorObject) : resolve(redirectOccurred);
-				newWindow.removeEventListener(closeListener);
-			};
-			newWindow.addEventListener("close", closeListener);
+						newWindow.removeEventListener(redirectListener);
+						newWindow.close();
```

Comment:
Yeah, I thought about that, though in the waitForClose option implies that we cannot monitor the results of the call (that or we don't care).  That being said, let me look at it again and see if it makes sense to still include the event listeners.